test/cluster/functionality.sh

```diff
@@ -48,14 +48,15 @@ elasticsearch_api="$( oc get svc "${OAL_ELASTICSEACH_SERVICE}" -o jsonpath='{ .m
 
 for kibana_pod in $( oc get pods --selector component="${OAL_KIBANA_COMPONENT}"  -o jsonpath='{ .items[*].metadata.name }' ); do
 	os::log::info "Testing Kibana pod ${kibana_pod} for a successful start..."
-	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Server running at http://0\.0\.0\.0:5601" "$(( 10*TIME_MIN ))"
-	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Kibana index ready" "$(( 10*TIME_MIN ))"
+	os::cmd::try_until_text "oc exec ${kibana_pod} -c kibana -- curl -s --request HEAD --write-out '%{response_code}' http://localhost:5601/" "200" "$(( 10*TIME_MIN ))"
+	os::cmd::expect_success_and_text "oc get pod ${kibana_pod} -o jsonpath='{ .status.containerStatuses[?(@.name==\"kibana\")].ready }'" "true"
+	os::cmd::expect_success_and_text "oc get pod ${kibana_pod} -o jsonpath='{ .status.containerStatuses[?(@.name==\"kibana-proxy\")].ready }'" "true"
 done
 
 for elasticsearch_pod in $( oc get pods --selector component="${OAL_ELASTICSEACH_COMPONENT}" -o jsonpath='{ .items[*].metadata.name }' ); do
 	os::log::info "Testing ElasticSearch pod ${elasticsearch_pod} for a successful start..."
-	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[cluster\.service\s*\]" "$(( 10*TIME_MIN ))"
-	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[node\s*\]\s*\[.*\]\s*started" "$(( 10*TIME_MIN ))"
+	os::cmd::try_until_text "oc exec ${elasticsearch_pod} -- curl -sk --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key -X HEAD -w '%{response_code}' https://localhost:9200/" '200' "$(( 10*TIME_MIN ))"
```

Comment:
It may take several minutes between curl returning 200 and the "cluster.service" and "node started" to appear in the logs, so this check is not nearly as strict, and possibly could report "ready" when it isn't so ready.  But I'm not really sure what to do about this, if there is some API call we can make that will return the same information.  Perhaps @portante or @lukas-vlcek can suggest something.

The name of the product is `Elasticsearch` not `ElasticSearch`.

Can you use this opportunity to change `ELASTICSEACH` to `ELASTICSEARCH`?